File: backend/src/movies/movies.controller.ts
import { Controller, Get, Query, ValidationPipe } from '@nestjs/common';
import { MoviesService } from './movies.service';
import { SearchMoviesDto } from './dto/search-movie.dto';
import { FetchMoviesDto } from './dto/fetch-movies.dto';

@Controller('movies')
export class MoviesController {
  constructor(private readonly moviesService: MoviesService) {}

  @Get()
  async fetchMovies(
    @Query(new ValidationPipe({ transform: true })) query: FetchMoviesDto,
  ) {
    return this.moviesService.fetchMovies(query?.genreId);
  }

  @Get('search')
  async searchMovies(
    @Query(new ValidationPipe({ transform: true })) query: SearchMoviesDto,
  ) {
    return this.moviesService.searchMovies(query.title);
  }
}

File: backend-java/src/main/java/com/laura/backendjava/controller/MoviesController.java
package com.laura.backendjava.controller;

import com.laura.backendjava.dto.Movie;
import com.laura.backendjava.service.MoviesService;
import java.util.List;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/movies")
public class MoviesController {
    private final MoviesService moviesService;

    public MoviesController(MoviesService moviesService) {
        this.moviesService = moviesService;
    }

    @GetMapping
    public List<Movie> fetchMovies(@RequestParam(required = false) Integer genreId) {
        return moviesService.fetchMovies(genreId);
    }

    @GetMapping("/search")
    public List<Movie> searchMovies(@RequestParam String title) {
        return moviesService.searchMovies(title);
    }
}